packages/expo-battery/android/src/main/java/expo/modules/battery/BatteryLevelReceiver.kt

```diff
@@ -0,0 +1,31 @@
+package expo.modules.battery
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.os.BatteryManager
+import android.os.Bundle
+import org.unimodules.core.interfaces.services.EventEmitter
+
+class BatteryLevelReceiver(private val eventEmitter: EventEmitter?) : BroadcastReceiver() {
+  private val BATTERY_LEVEL_EVENT_NAME = "Expo.batteryLevelDidChange"
+
+  private fun onBatteryLevelChange(BatteryLevel: Float) {
+    eventEmitter?.emit(BATTERY_LEVEL_EVENT_NAME, Bundle().apply {
+      putFloat("batteryLevel", BatteryLevel)
+    })
+  }
+
+  override fun onReceive(context: Context, intent: Intent) {
+    val batteryIntent = context.applicationContext.registerReceiver(null, IntentFilter(Intent.ACTION_BATTERY_CHANGED))
+    val level = batteryIntent!!.getIntExtra(BatteryManager.EXTRA_LEVEL, -1)
```

Comment:
There is no guarantee that the `batteryIntent` won't be `null`, so I'd recommend unwrapping it and maybe print an error if it's not available. Generally I would advise not to use `!!` 😉